### PR TITLE
[JENKINS-44266, JENKINS-36654] Fix PCT for 2.46.2

### PR DESCRIPTION
**Pull request by varyvol**
[JENKINS-44266](https://issues.jenkins-ci.org/browse/JENKINS-44266)

- Bump core version.
- Bump some plugins versions.
- Remove HudsonTestCase and use JenkinsRule.

@reviewbybees @oleg-nenashev @kwhetstone

### CODE REVIEW

**Comment by varyvol** [comment]
@jglick @oleg-nenashev it's the usual issue with transitive test dependencies. I have improved the JIRA issue with the actual error message.

**Comment by jglick** [comment]
So does https://github.com/jenkinsci/plugin-compat-tester/pull/28 fix it?

**Comment by varyvol** [comment]
@jglick no, that would avoid a transitive optional dependency to be added to the POM, but this is failing even without modifying the POM (check that the command just changes the core version).

**Comment by jglick** [comment]
I thought I saw that this plugin was not in the `dependency:tree` but I forget now.

**Comment by varyvol** [comment]
@oleg-nenashev @jglick do you think this need further work?

**Comment by varyvol** [comment]
@oleg-nenashev could you review this again?

**Comment by varyvol** [comment]
@oleg-nenashev I will look for it, don't worry.

**Comment by varyvol** [comment]
@oleg-nenashev done and yes, adding subversion is not necessary now.
@jglick it would be great if you could review this again.

File: pom.xml [hunk 1]
```diff
@@ -47,43 +47,11 @@
     </license>
   </licenses>
 
-  <repositories>
```
Comment (oleg-nenashev) [inline pom.xml]: 🐛 In such case it won't be able to retrieve Parent pom if it is missing in the local cache.

OTOH you can simplify it to
```xml
    <repositories> 
      <repository>
        <id>repo.jenkins-ci.org</id>
        <url>http://repo.jenkins-ci.org/public/</url>
      </repository>
    </repositories>
    <pluginRepositories>
      <pluginRepository>
        <id>repo.jenkins-ci.org</id>
        <url>http://repo.jenkins-ci.org/public/</url>
      </pluginRepository>
    </pluginRepositories>
```

Comment (jglick) [inline pom.xml]: Well, use `https` but otherwise yes that is the standard blurb.

Comment (oleg-nenashev) [inline pom.xml]: copypasted from the wrong plugin :(

Comment (varyvol) [inline pom.xml]: My bad, I thought this was already included in the parent POM.

**Comment by varyvol** [comment]
@oleg-nenashev I have added back the "repositories" tag and replaced @Bug for @Issue.

File: src/test/java/hudson/plugins/promoted_builds/ConfigurationDoCheckTest.java [hunk 2]
```diff
@@ -22,7 +21,7 @@
     @Rule
     public JenkinsRule j = new JenkinsRule();
 
-    @Bug(7972)
+    @Issue("7972")
```
Comment (oleg-nenashev) [inline src/test/java/hudson/plugins/promoted_builds/ConfigurationDoCheckTest.java]: Please use full id, e.g. "JENKINS-7972"

Comment (varyvol) [inline src/test/java/hudson/plugins/promoted_builds/ConfigurationDoCheckTest.java]: Done.

**Comment by oleg-nenashev** [comment]
It also fixes https://issues.jenkins-ci.org/browse/JENKINS-36654 from what I see

**Comment by oleg-nenashev** [comment]
Merging. Will see if I can release it today

**Pull request merged**